### PR TITLE
Update LocalVariableTypeTable for generic type of method arguments or local variables.

### PR DESCRIPTION
**Pull request by KyongSik-Yoon**
If a method for BCI have at least one generic type of argument JVM throw java.lang.ClassFormatError due to not updated LocalVariableTable.

I know there is workaround as call removeLocalVariables method.
But I think it's better that BCEL can stay proper LocalVariableTable.

I'm also pull test case can reproduce error.

### CODE REVIEW

**Comment by KyongSik-Yoon** [comment]
Attach error stack.

java.lang.ClassFormatError: LVTT entry for 'a2' in class file org/apache/bcel/data/SimpleClassHasMethodIncludeGenericArgument does not match any LVT entry

	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
	at org.apache.bcel.LocalVariableTypeTableTest$TestClassLoader.findClass(LocalVariableTypeTableTest.java:27)
	at org.apache.bcel.LocalVariableTypeTableTest.testWihtoutGenericArgument(LocalVariableTypeTableTest.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:117)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:253)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)

**Comment by mebigfatguy** [comment]
Thanks alot! Can you please add a ticket to the bcel jira here: http://issues.apache.org/jira/browse/BCEL

**Comment by KyongSik-Yoon** [comment]
I added a ticket to Jira. 
https://issues.apache.org/jira/browse/BCEL-276

**Comment by KyongSik-Yoon** [comment]
I added more modification for the other bug.
A method that constantToString of ConstantPool class always occur NoSuchElementException when face with costant type of CONSTANT_NameAndType.

A reason is invalid separator for StringTokenizer that colon. It should be replaced to empty string.
This pull request include ConstantPoolTestCase.

**Comment by KyongSik-Yoon** [comment]
https://issues.apache.org/jira/browse/BCEL-277

File: src/main/java/org/apache/bcel/generic/MethodGen.java [hunk 1]
```diff
@@ -25,21 +25,7 @@
...
-import org.apache.bcel.classfile.LineNumber;
-import org.apache.bcel.classfile.LineNumberTable;
-import org.apache.bcel.classfile.LocalVariable;
-import org.apache.bcel.classfile.LocalVariableTable;
-import org.apache.bcel.classfile.Method;
-import org.apache.bcel.classfile.ParameterAnnotationEntry;
-import org.apache.bcel.classfile.ParameterAnnotations;
-import org.apache.bcel.classfile.RuntimeVisibleParameterAnnotations;
-import org.apache.bcel.classfile.Utility;
+import org.apache.bcel.classfile.*;
```
Comment (britter) [inline src/main/java/org/apache/bcel/generic/MethodGen.java]: Please don't use wild card imports

File: src/test/java/org/apache/bcel/LocalVariableTypeTableTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
+package org.apache.bcel;
```
Comment (britter) [inline src/test/java/org/apache/bcel/LocalVariableTypeTableTest.java]: Please add the Apache License v2 License header

File: src/test/java/org/apache/bcel/classfile/ConstantPoolTestCase.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package org.apache.bcel.classfile;
```
Comment (britter) [inline src/test/java/org/apache/bcel/classfile/ConstantPoolTestCase.java]: Please add the Apache License v2 Header

File: src/test/java/org/apache/bcel/data/SimpleClassHasMethodIncludeGenericArgument.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package org.apache.bcel.data;
```
Comment (britter) [inline src/test/java/org/apache/bcel/data/SimpleClassHasMethodIncludeGenericArgument.java]: Please add the Apache License v2 Header

**Comment by britter** [comment]
How does this PR relate to https://issues.apache.org/jira/browse/BCEL-276 ?

**Comment by KyongSik-Yoon** [comment]
It's my mistake. I didn`t divide two issues.  
So this commit contains two kind of the bugs that BCEL-276 and BCEL-277. 

I`ll pull request again after divide it with fix for  your comment.

**Comment by britter** [comment]
Okay, cool! Ping me when you're ready. If you intend to contribute to Commons BCEL on a regular basis you should also sign an ICLA as described in CONTRIBUTING.md.

Thank you!

**Comment by KyongSik-Yoon** [comment]
@britter I'm ready now. I registered two PR for that BCEL-276 and BCEL-277. Please check it.

**Comment by britter** [comment]
@KyongSik-Yoon Thank you! I've merged #9 and commented on #10. I think this PR can be closed.

**Comment by KyongSik-Yoon** [comment]
Close this PR.

**Pull request closed without merging**